### ISSUE TITLE
Error accessing file: file not found

### ISSUE THREAD

**Issue by joelfgs** [github/git-lfs#645]
A coworker pushed a large set of binary files last Friday but I am not able to download them.  When I try to pull I get:

```
Error accessing media: path/to/file/mypngfile.png (57f4fb5241459e98b44ca8c013ddf88dcfc8bf02871e4a1744e44fb38b7a669a)

Errors logged to .git/lfs/objects/logs/20150908T115221.040467423.log
Use `git lfs logs last` to view the log.
```

and the lfs log shows:

```
> git-lfs smudge path/to/file/mypngfile.png
LocalWorkingDir=/Users/me/Work/repositoryname
LocalGitDir=.git
LocalMediaDir=.git/lfs/objects
TempDir=.git/lfs/tmp
ConcurrentTransfers=3
BatchTransfer=false
GIT_INTERNAL_GETTEXT_SH_SCHEME=fallthrough
GIT_DIR=.git
GIT_REFLOG_ACTION=pull
GIT_PREFIX=

Error accessing media: path/to/file/mypngfile.png (57f4fb5241459e98b44ca8c013ddf88dcfc8bf02871e4a1744e44fb38b7a669a)

Error downloading .git/lfs/objects/57/f4/57f4fb5241459e98b44ca8c013ddf88dcfc8bf02871e4a1744e44fb38b7a669a.
Not Found
Docs: https://github.com/contact
Endpoint=https://github.com/organizationname/repositoryname/info/lfs
Request:Access-Control-Allow-Credentials=true
Request:Server=GitHub.com
Request:Content-Type=application/json; charset=utf-8
Response:Authorization=--
Request:X-Ratelimit-Remaining=2956
Request:Access-Control-Expose-Headers=ETag, Link, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval
Request:X-Ratelimit-Reset=1441737536
Request:Strict-Transport-Security=max-age=31536000; includeSubdomains; preload
Request:Access-Control-Allow-Origin=*
Request:X-Github-Request-Id=18074376:7F5F:5325039:55EF2B23
Request:X-Content-Type-Options=nosniff
Request:Date=Tue, 08 Sep 2015 18:38:28 GMT
Request:Content-Security-Policy=default-src 'none'
Request:X-Ratelimit-Limit=3000
Request:X-Xss-Protection=1; mode=block
Request:X-Frame-Options=deny
URL=GET https://github.com/organizationname/repositoryname/info/lfs/objects/57f4fb5241459e98b44ca8c013ddf88dcfc8bf02871e4a1744e44fb38b7a669a
Response:User-Agent=git-lfs/0.5.3 (GitHub; darwin amd64; go 1.3)
Response:Accept=application/vnd.git-lfs+json; charset=utf-8
Status=404 Not Found
Request:Status=404 Not Found
Request:X-Github-Media-Type=unknown
goroutine 16 [running]:
github.com/github/git-lfs/lfs.Stack(0x0, 0x0, 0x0)
	/Users/rick/go/src/github.com/github/git-lfs/lfs/errors.go:85 +0x89
github.com/github/git-lfs/lfs.Errorf(0x692e38, 0xc2082a5590, 0x3b6fc0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/rick/go/src/github.com/github/git-lfs/lfs/errors.go:29 +0x62
github.com/github/git-lfs/lfs.Error(0x692e38, 0xc2082a5590, 0x5bf3c8)
	/Users/rick/go/src/github.com/github/git-lfs/lfs/errors.go:17 +0x52
github.com/github/git-lfs/lfs.handleResponse(0xc208047cb0, 0x0)
	/Users/rick/go/src/github.com/github/git-lfs/lfs/client.go:495 +0x18e
github.com/github/git-lfs/lfs.doHttpRequest(0xc2080000d0, 0xc20800ff20, 0x100000002, 0x2)
	/Users/rick/go/src/github.com/github/git-lfs/lfs/client.go:377 +0x406
github.com/github/git-lfs/lfs.doApiRequestWithRedirects(0xc2080000d0, 0xc20800ff20, 0x6c87d0, 0x0, 0x4, 0x0, 0x0)
	/Users/rick/go/src/github.com/github/git-lfs/lfs/client.go:392 +0x44
github.com/github/git-lfs/lfs.doApiRequest(0xc2080000d0, 0xc20800ff20, 0xc208005097, 0x40, 0xc2080000d0)
	/Users/rick/go/src/github.com/github/git-lfs/lfs/client.go:441 +0x8e
github.com/github/git-lfs/lfs.Download(0xc208005097, 0x40, 0x0, 0x0, 0x6c8978, 0x2)
	/Users/rick/go/src/github.com/github/git-lfs/lfs/client.go:112 +0xd9
github.com/github/git-lfs/lfs.downloadFile(0x687318, 0xc208038008, 0xc20803a440, 0x7fff5fbff697, 0x3e, 0xc208005080, 0x57, 0x0, 0x0)
	/Users/rick/go/src/github.com/github/git-lfs/lfs/pointer_smudge.go:112 +0x1e8
github.com/github/git-lfs/lfs.PointerSmudge(0x687318, 0xc208038008, 0xc20803a440, 0x7fff5fbff697, 0x3e, 0x0, 0x0, 0x0)
	/Users/rick/go/src/github.com/github/git-lfs/lfs/pointer_smudge.go:32 +0x30d
github.com/github/git-lfs/lfs.(*Pointer).Smudge(0xc20803a440, 0x687318, 0xc208038008, 0x7fff5fbff697, 0x3e, 0x0, 0x0, 0x0)
	/Users/rick/go/src/github.com/github/git-lfs/lfs/pointer.go:44 +0x73
github.com/github/git-lfs/commands.smudgeCommand(0x5b9420, 0xc208034d30, 0x1, 0x1)
	/Users/rick/go/src/github.com/github/git-lfs/commands/command_smudge.go:60 +0x819
github.com/github/git-lfs/vendor/_nuts/github.com/spf13/cobra.(*Command).execute(0x5b9420, 0xc208034d20, 0x1, 0x1, 0x0, 0x0)
	/Users/rick/go/src/github.com/github/git-lfs/vendor/_nuts/github.com/spf13/cobra/command.go:355 +0x17c
github.com/github/git-lfs/vendor/_nuts/github.com/spf13/cobra.(*Command).findAndExecute(0x5b9c20, 0xc20800e010, 0x2, 0x2, 0x0, 0x0)
	/Users/rick/go/src/github.com/github/git-lfs/vendor/_nuts/github.com/spf13/cobra/command.go:334 +0xd3
github.com/github/git-lfs/vendor/_nuts/github.com/spf13/cobra.(*Command).Execute(0x5b9c20, 0x0, 0x0)
	/Users/rick/go/src/github.com/github/git-lfs/vendor/_nuts/github.com/spf13/cobra/command.go:408 +0x7b0
github.com/github/git-lfs/commands.Run()
	/Users/rick/go/src/github.com/github/git-lfs/commands/commands.go:84 +0x2a
main.main()
	/Users/rick/go/src/github.com/github/git-lfs/git-lfs.go:9 +0x1a
```

I checked with the coworker and he said that when he pushed there didn't seem to be any `lfs` activity - git appeared to do a simple push without uploading any files.  I can browse the repository and see what look like marker files for `LFS` that were put there last Friday by the push, contents like:

```
version https://git-lfs.github.com/spec/v1
oid sha256:d6e4388e96d6d3ae5970696987e0ca6de26c812d6711bf93bf844576f499af6c
size 6971
```

We're not sure what to do at this point.  `lfs` seems to know about the files but within his local repository it thinks they've all been uploaded, and from mine they don't exist.

**Comment by technoweenie** [comment]
He'll have to push the files with `git lfs push origin master` (replace "origin" and "master" as needed). Typically, this is handled automatically through a git pre-push. Does he have it installed? You can check by running this from the git repository that he's pushing from:

```
$ cat .git/hooks/pre-push
#!/bin/sh
command -v git-lfs >/dev/null 2>&1 || { echo >&2 "\nThis repository is configured for Git LFS but 'git-lfs' was not found on your path. If you no longer wish to use Git LFS, remove this hook by deleting .git/hooks/pre-push.\n"; exit 2; }
git lfs pre-push "$@"
```

That hook should push the LFS objects to the server _before_ pushing commits to the Git server.

**Comment by lclibardi** [comment]
Hi, the coworker here, the pre-push hook is there, slightly different wording but there:
```
#!/bin/sh
command -v git-lfs >/dev/null 2>&1 || { echo >&2 "\nThis repository has been set up with Git LFS but Git LFS is not installed.\n"; exit 0; }
git lfs pre-push "$@"
```

And `git lfs -h` displays the help as expected.  Tried running `git lfs push origin master` with to re-push the files but it responds with:  `(0 of 0 files) 0 B 0`

**Comment by technoweenie** [comment]
What about `git-lfs`? What OS are you on?

**Comment by lclibardi** [comment]
Running `git-lfs` displays the Usage and Available commands.
I'm on Ubuntu 14.04 LTS.

**Comment by technoweenie** [comment]
Is it possible that git lfs was not installed correctly when you pushed? The `pre-push` script that you have has a known issue. That message exits with `0`, which causes the git push to go ahead without uploading the LFS objects. This is fixed in v0.6.0, which is coming out in the next day or so.

If you have Git LFS v0.5.4, you can fix it with:

```
# git lfs push --object-id origin [oid] [oid] [oid]
$ git lfs push --object-id origin 57f4fb5241459e98b44ca8c013ddf88dcfc8bf02871e4a1744e44fb38b7a669a
```

This works, but requires you to put the object OIDs in the path manually. `git lfs push` didn't work because your local clone of the remote has been updated from your successful push. #646 adds a `--all` flag that will push everything: `git lfs push --all origin master`

**Comment by sinbad** [comment]
This could have been caused by pushing multiple branches at once, see #632 - any branches after the first to the `git push` command would have been ignored. This is fixed for 0.6.

**Comment by EverNewJoy** [comment]
**2018 Solution for git-lfs: command not found**
_git-lfs smudge -- git-lfs: command not found_


I ran into this issue using git for windows desktop, I had already installed git LFS and was using it just fine.

But when I finally had to revert to a previous commit using git reset --hard <commit>, all of sudden I got the error message git-lfs: command not found

The solution for me was to copy the git-lfs.exe to the location of git.exe (windows 10 machine) into the program files/git/bin folder 

Hope this helps someone 

♥
